server/utils/authMiddleWare.js

```diff
@@ -0,0 +1,21 @@
+import { firebaseAdmin } from '../database/firestore';
+
+export function authMiddleWare(req, res, next) {
+  const authHeader = req.headers.authorization;
+  if (!authHeader) {
+    return res.send({ message: 'Token Missing.' }).status(401);
+  }
+
+  if (authHeader && authHeader.split(' ')[0] !== 'Bearer') {
+    res.send({ message: 'Invalid token' }).status(401);
+  }
+
+  const token = authHeader.split(' ')[1];
+  firebaseAdmin
+    .auth()
+    .verifyIdToken(token)
+    .then(() => next())
+    .catch(() => res.send({ message: 'Authorization failed' }).status(403));
+
+  return token;
```

Comment:
We'll need to call `next()` if this is a middleware

I'd also recommend we fetch the user and add it to the request

Reply:
The linter was screaming about the function not having a return value so I just had it return token. So call next() before that?

Reply:
We can disable that rule then, but it's because you have:
```
    return res.send({ message: 'Token Missing.' }).status(401);
```

If you didn't return any of them and ensured the if/elses were separate, this would work. Right now, the 401 invalid token would be sent but the rest of the actions would still be executed